File: backend/utils/progress.go
package utils

func CalculateProgress(theory, examples bool, testScore int) int {
	progress := 0

	if theory {
		progress += 20
	}
	if examples {
		progress += 30
	}

	if theory && examples {
		progress += int(float64(testScore) / 20 * 50)
	} else if testScore > 0 {
		progress += 50
	}

	if progress > 100 {
		progress = 100
	}

	return progress
}

File: frontend/js/progress.js
function updateProgress(course, theory=false, examples=false, testScore=0) {
    const token = localStorage.getItem("token");
    if (!token) {
        alert("Сначала войдите в систему!");
        return;
    }

    fetch("https://YOUR_RENDER_URL/api/progress/update", {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
            "Authorization": "Bearer " + token
        },
        body: JSON.stringify({ course, theory, examples, testScore })
    })
    .then(res => res.json())
    .then(data => alert(`Прогресс обновлён для ${course}`))
    .catch(err => console.error(err));
}
